### ISSUE TITLE
Move each pluginbridge into its own maven module

### ISSUE THREAD

**Issue by Phoenix616** [Rsl1122/Plan-PlayerAnalytics#868]
### Is your feature request related to a problem? Please describe.
Currently the PluginBridge module requires all plugins it depends on to be available in order to build it which might result in build errors when certain repositories go down or when you don't actually need support for certain plugins and don't want to build support for them. In that case you have to manually edit the source instead of just not building the module. It also increases the build time.

### Describe the solution you'd like
- [ ] 1. Create maven module for each plugin
- [ ] 2. Optionally add a dynamic loading of bridges for specific plugins so that you don't even need to edit a pom if you don't want to include certain modules but just don't build/add them to the plugin.

### Describe alternatives you've considered
Editing code manually which gets annoying the more plugins get supported.

**Comment by Rsl1122** [comment]
Hello.
As of 4.5.2, PlanPluginBridge is distributed via Bintray maven repository, and does not need to be compiled by other developers in order to build Plan.

### Intention of the PluginBridge:

1. 3rd party plugins may not provide a maven repository, so including these as dependencies directly from Plan would make contributions to the main project troublesome. Some plugins that are supported are paid plugins and might not provide a separate API artifact
2. Hooking into 3rd party plugins out-of-the-box
3. *(No longer needed) Need to install only a single artifact manually to build Plan (Now automatically installed via distribution)*

I'm trying to figure out why you would like to build the plugin bridge

#### If you want to build custom version of Plan that supports your plugin

I recommend that instead of building Plan, build a separate project that depends on both Plan and the plugin, using the provided API https://github.com/Rsl1122/Plan-PlayerAnalytics/wiki/APIv4

Another option is adding Plan support to the plugin, although Plan API is not supplied in any maven repository. (Maybe when v5 arrives)

#### If you want to reduce the size of Plan.jar

The plugin bridge is 152kb while Plan is 9Mb. Instead I recommend taking out some project dependencies that are not in use (H2 for example).

#### Other reason

Please comment below why you would like to build the plugin bridge

**Comment by Phoenix616** [comment]
Ah, I see, I somehow was under the impression that the PluginBridge was always built alongside the core plugin and not a completely different library.

I still think that different modules or another dynamic approach could simplify contributing to the PluginBridge as one wouldn't have to somehow acquire all of the plugins without a repository/premium ones or manually remove all references to them when testing ones additions to the bridge.

**Comment by Rsl1122** [comment]
I'll try to look at solutions other plugins have come up with for this issue and get back to you.

**Comment by Rsl1122** [comment]
I'll close this issue since I feel like it has been resolved. Here is what I found

#### Bentobox

Bentobox has an add-on API and each add-on is maintained in its own maven project and git repository.
There is no cyclic dependency like in the case of Plan, as the main Bentobox artifact does not include the add-ons. The add-ons are loaded separately.

#### PlaceholderAPI

PlaceholderAPI has an extension api and each extension is maintained in its own maven project and git repository.
This system is very similar to the one in use on Bentobox and the extensions are loaded via reflection.

----

> Ah, I see, I somehow was under the impression that the PluginBridge was always built alongside the core plugin and not a completely different library.

I think to avoid confusion wise thing to do would be to split PlanPluginBridge folder to its own git repository.

----

I'll add this to be considered when writing the new plugin data API. (#638)

Fundamental problem here is that Plan and PlanPluginBridge are cyclic dependencies of each other (Plan depends on the bridge and bridge depends on Plan for the API).

Others have solved this by not including the extensions in the main artefact, which is the sane thing to do, but in the case of Plan inconvenient to the users. 

#### Possible solution

I can solve the inconvenience same way others have, by loading the PluginData providers via reflection, eliminating the cyclic dependency. 

Here is how the module dependencies will look after the solution is implemented.

![plan-modules](https://user-images.githubusercontent.com/24460436/50647515-6c14a180-0f81-11e9-98de-ce6af47e1dc5.jpg)
